python/api/main.py

```diff
@@ -167,9 +167,26 @@ async def get_hfp_points(stop_id: str):
                 )
 
             observation_geojson_features = []
-
             for observation_tuple in totalObservations:
                 observation = observation_tuple[0]
                 observation_geojson_features.append(observation)
 
     return get_feature_collection(observation_geojson_features)
+
+
+@app.get("/percentile_circles/{stop_id}")
+async def get_percentile_circles(stop_id: str):
+    """
+    Returns a GeoJSON FeatureCollection of percentile circles with given stop_id.
+    """
+    with psycopg.connect(**get_conn_params()) as conn:
+        with conn.cursor() as cur:
+            cur.execute("SELECT api.get_percentile_circles_with_stop_id(%(stop_id)s)", {'stop_id': stop_id })
+            percentile_circles = cur.fetchall()
+
+            percentile_circle_geojson_features = []
```

Comment:
Seems like we have a repeated pattern here, common for all these endpoints returning a FeatureCollection. At the same time, the current `get_feature_collection()` func does pretty little compared to this pattern. Maybe we could wrap the whole thing to a function in `common.utils`, e.g. like this?

```python
def tuples_to_feature_collection(geom_tuples: list[tuple]) -> dict
    """Transforms GeoJSON tuples returned by psycopg into a FeatureCollection dict for REST API."""
    # Effectively the same as the list initialization + for loop phase:
    features = [tp[0] for tp in geom_tuples]
    # Then what the current get_feature_collection() does (making that function obsolete now):
    return {
        "type": "FeatureCollection",
        "features": features
    }
```

which would be used here like

```python
return tuples_to_feature_collection(percentile_circles)
```